backend/controllers/cafeController.js

```diff
@@ -1,14 +1,47 @@
-const Cafe = require('../models/cafeModel')
+const Cafe = require('../models/cafeModel');
+const defaultDistance = 5000; // in meters, i.e., 5000 = 5km
 
 exports.getCafesByFacilitiesAndLocation = async (req, res, next) => {
-  const { facilities, requiredFacilities } = req.query;
-  const requestedFacilities = requiredFacilities ? requiredFacilities : facilities;
-  const requestedFacilitiesArray = requestedFacilities.split(',');
+  const { facilities, requiredFacilities, userLocation } = req.query;
+  const maxDistance = Number(req.query.distance) || defaultDistance;
+
+  if ((!requiredFacilities && !facilities) || !userLocation) {
+    throw new Error('Both facilities and location are required.');
+  }
+
+  const facilitiesQuery = requiredFacilities
+    ? { $all: requiredFacilities.split(',') }
```

Comment:
This looks very cool! I'd be interested in a run-through or documentation of what you used to action this